### PR TITLE
Refactors some plasmacutter behavior, squashes a number of plasmacutter bugs

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Refactors plasmacutter behavior when used on walls, fixing a number of bugs (see changelog for details) and making the code more concise. Also addresses and fixes https://github.com/tgstation/TerraGov-Marine-Corps/issues/9468.

## Why It's Good For The Game

Bugs are bad, fixes good.

## Changelog
:cl:
balance: Plasmacutters will now cut down walls without leaving a girder
fix: Plasmacutters now respect resistance flags and cannot destroy turfs that have RESIST_ALL or PLASMACUTTER_IMMUNE. 
fix: Plasmacutters no longer spam "can't cut through this" messages when destroying some closed turfs.
fix: Fixes a rare situation where plasmacutters would take multiple charges to cut down some ship walls.
refactor: Refactored plasmacutter behavior for closed turfs.
/:cl:

### CODE REVIEW

File: code/game/turfs/closed.dm [hunk 1]
```diff
@@ -168,27 +168,33 @@
 
 	if(istype(I, /obj/item/tool/pickaxe/plasmacutter) && !user.do_actions)
 		var/obj/item/tool/pickaxe/plasmacutter/P = I
-		if(!ismineralturf(src) && !istype(src, /turf/closed/gm/dense) && !istype(src, /turf/closed/glass) && !istype(src, /turf/closed/desertdamrockwall) && !istype(src, /turf/closed/brock))
+		if(CHECK_BITFIELD(resistance_flags, RESIST_ALL) || CHECK_BITFIELD(resistance_flags, PLASMACUTTER_IMMUNE))
 			to_chat(user, span_warning("[P] can't cut through this!"))
 			return
-		if(!P.start_cut(user, name, src))
+		if(istype(src, /turf/closed/wall)) //walls handle plasma cutter effects on their own
 			return
```
Comment (TiviPlus) [inline code/game/turfs/closed.dm:175]: istype src is bad

File: code/game/turfs/closed.dm [hunk 2]
```diff
@@ -168,27 +168,33 @@
...
+		else if(istype(src, /turf/closed/ice) || istype(src, /turf/closed/ice_rock))
+			ChangeTurf(/turf/open/floor/plating/ground/ice)
 		else if(istype(src, /turf/closed/brock))
 			var/choice = rand(1,50)
 			if(choice == 50)
 				ChangeTurf(/turf/open/lavaland/basalt/glowing)
 			else
 				ChangeTurf(/turf/open/lavaland/basalt)
+		else if(!istype(src, /turf/closed/wall)) //walls handle deconstruction on their own so don't apply plating
+			ChangeTurf(/turf/open/floor/plating) //if none of the above apply regular plating as a fallback
```
Comment (TiviPlus) [inline code/game/turfs/closed.dm:197]: yuck but not your fault

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/turfs/closed.dm [hunk 3]
```diff
@@ -166,29 +166,44 @@
 /turf/closed/attackby(obj/item/I, mob/user, params)
 	. = ..()
 
+	var/turf/targettedwall = get_turf(src)
+
```
Comment (TiviPlus) [inline code/game/turfs/closed.dm:170]: there can only be one turf on a turf

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/turfs/closed.dm [hunk 4]
```diff
@@ -168,27 +168,40 @@
...
-		else if(istype(src, /turf/closed/gm/dense))
+
+		else if(isjungleturf(src))
 			ChangeTurf(/turf/open/ground/jungle/clear)
-		else if(istype(src, /turf/closed/brock))
+
+		else if(isiceturf(src) || isicerockturf(src))
+			ChangeTurf(/turf/open/floor/plating/ground/ice)
+
+		else if(isvolcanicturf(src))
```
Comment (BraveMole) [inline code/game/turfs/closed.dm:196]: No no no :).

you need to add a var to turfs (let's call it open_turf_type) 
and then you do ChangeTurf(open_turf_type)

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Pull request merged**